### PR TITLE
[FIX] payment_worldline: merchantReference is limited to 30 characters

### PR DESCRIPTION
**Pull request by xavieralt**
The hostedcheckout param  `order.references.merchantReference` is limited to 30 characters [^1], so we must not exceed that size for wordline's payment.transaction reference.

[^1]: https://apireference.connect.worldline-solutions.com/s2sapi/v1/en_US/go/hostedcheckouts/create.html?paymentPlatform=ALL

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_worldline/models/payment_transaction.py [hunk 1]
```diff
@@ -19,6 +19,17 @@
 class PaymentTransaction(models.Model):
     _inherit = 'payment.transaction'
 
+    def _compute_reference(self, provider_code, prefix=None, separator='-', **kwargs):
+        ref = super()._compute_reference(provider_code, prefix=prefix, separator=separator, **kwargs)
+        if provider_code != 'worldline':
+            return super()._compute_reference(provider_code, prefix=prefix, separator=separator, **kwargs)
```
Comment (Demesmaeker) [inline addons/payment_worldline/models/payment_transaction.py:25]:
```suggestion
            return super()._compute_reference(
                provider_code, prefix=prefix, separator=separator, **kwargs
            )
```
Nitpicking: we try to stay < 100 chars per line =]

**Review by Demesmaeker — CHANGES_REQUESTED** [review]
Hello @xavieralt 
Thanks for your fix, this'd ensure the prefix is not longer than 30 chars, but not that the merchantReference is also limited to 30 chars, if that reference already exists, it'll be completed like here: https://github.com/odoo/odoo/blob/18.0/addons/payment/models/payment_transaction.py#L381
Have a nice day,

**Comment by xavieralt** [comment]
@Demesmaeker
- fixed lines exceeding 100+ chars
- for the reference: we try computing it the normal way (which will works most of the time), if not we compute a generic reference with `WL` prefix (which will generate a 17-chars reference when singularized, so we have enough space left for the sequence counter)

**Review by Demesmaeker — APPROVED** [review]
Thanks for your changes ^^

**Comment by Demesmaeker** [comment]
@robodoo r+

**Pull request closed without merging**